### PR TITLE
BUG: positional getitem indexing with list on Series with duplicate integer index fails 

### LINKED ISSUE #17347: BUG: positional getitem indexing with list on Series with duplicate integer index fails
#### Problem description

With a Series with integer index, `__getitem__` / [] does positional indexing, but when you have a duplicate index and use a list indexer, you get a "ValueError: cannot reindex from a duplicate axis" (with `.loc` the same works OK)

#### Code Sample, a copy-pastable example if possible

```python
In [28]: s = pd.Series(range(3), index=[1, 2, 3])

In [29]: s[1]
Out[29]: 0

In [30]: s[[1]]
Out[30]: 
1    0
dtype: int64

# duplicate index
In [31]: s = pd.Series(range(3), index=[1, 1, 3])

In [32]: s[1]
Out[32]: 
1    0
1    1
dtype: int64

In [33]: s[[1]]
...
ValueError: cannot reindex from a duplicate axis

In [34]: s.loc[[1]]
Out[34]: 
1    0
1    1
dtype: int64
```

It seems to only fail for integer index (eg with a string index `s = pd.Series(range(3), index=['a', 'a', 'b'])` the equivalent works ok)

### PR DESCRIPTION
**Pull request by tranctan**
Create test case for issue GH17347

- [x] closes #17347 
- [x] tests added / passed

### CODE REVIEW

File: pandas/tests/indexing/test_indexing.py [hunk 1]
```diff
@@ -623,6 +623,27 @@ def test_index_type_coercion(self):
...
+
+        # Check if s contains duplicated idx
+        if not dup_idx_filter.any:
+            pass
+        # Find the duplicate indexes
+        dup_idxes = s[dup_idx_filter].index.unique()
+
+        # Assert the duplicate indexes
+        for idx in dup_idxes:
+            assert s[idx].equals(s[[idx]])
```
Comment (mroeschke) [inline pandas/tests/indexing/test_indexing.py]: Use `tm.assert_series_equal`

File: pandas/tests/indexing/test_indexing.py [hunk 2]
```diff
@@ -623,6 +623,27 @@ def test_index_type_coercion(self):
...
+    @pytest.mark.parametrize('idx', [
+        Index([1, 1, 3]),
+        Index(['a', 'a', 'b', 'c', 'c', 'd']),
+        Index([1, 1, 1, 2, 2, 3]),
+        Index([1, 1, 2, 3, 1, 2, 3, 3])
+    ])
+    def test_duplicate_int_indexing(self, idx):
+        # GH 17347
+        s = pd.Series(range(len(idx)), idx)
+        dup_idx_filter = s.index.duplicated(keep=False)
```
Comment (mroeschke) [inline pandas/tests/indexing/test_indexing.py]: What's the purpose for all this `duplicated` logic? You can just copy the example from the original issue.

**Review by mroeschke — CHANGES_REQUESTED** [review]
Some comments.

File: pandas/tests/indexing/test_indexing.py (hunk 2 above)
Comment (tranctan96) [reply pandas/tests/indexing/test_indexing.py]: Hi ! Thank you for your review. This is my first time making a test case contribution, so i just figured it out that i overdid it. I'm fixing it now!

File: pandas/tests/indexing/test_indexing.py [hunk 3]
```diff
@@ -623,6 +623,10 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_duplicate_int_indexing(self):
+        # GH 17347
+        s = pd.Series(range(3), index=[1, 1, 3])
+        tm.assert_series_equal(s[1], s[[1]])
```
Comment (mroeschke) [inline pandas/tests/indexing/test_indexing.py]: Can you also test `loc`?

And format the test like

```
result = ...
expected = ...
tm.assert_series_equal(result, expected)
```

File: pandas/tests/indexing/test_indexing.py [hunk 4]
```diff
@@ -623,6 +623,15 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_duplicate_int_indexing(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: use parameterize like

```
@pytest.parameterize('case', [lambda s: s.__getitem__, lambda: s.loc])
def test_duplicate_int_indexing(self, case):
     expected = s.iloc[[1]]
     result = case(s)[[1]]
````

File: pandas/tests/indexing/test_indexing.py [hunk 5]
```diff
@@ -623,6 +623,15 @@ def test_index_type_coercion(self):
...
+        # GH 17347
+        s = pd.Series(range(3), index=[1, 1, 3])
+        expected = s[1]
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: move this after ``def test_dups_fancy_indexing2(self)``

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @tranctan96

**Pull request merged**